### PR TITLE
Add more logs for debugging

### PR DESCRIPTION
**Pull request by karlkfi**
- Rename "unmanage" in the remediator (internal enum & log msgs)  to "abandon" to be more consistent with the language used by the applier. These are both the same thing. It's when the object is removed from the inventory and its management annotations are removed (`client.lifecycle.config.k8s.io/deletion: detach`, `cli-utils.sigs.k8s.io/on-remove: keep`).
- Add logs before status updates to indicate when/why

### CODE REVIEW

File: pkg/diff/diff.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ const (
 	// Error indicates the resource's management annotation in the API server is invalid.
 	Error = Operation("error")
 
-	// Unmanage indicates the resource's management annotation should be removed from the API Server.
-	Unmanage = Operation("unmanage")
+	// Abandon indicates the resource's management annotation should be removed from the API Server.
```
Comment (nan-yu) [inline pkg/diff/diff.go:56]: Are we going to distinguish between abandon and unmanage?

Comment (karlkfi) [reply pkg/diff/diff.go:56]: Is there a difference?
Abandoning is removing it from the inventory and removing the management annotations.

Comment (nan-yu) [reply pkg/diff/diff.go:56]: If no difference, I'm curious why we need to rename it.

Comment (karlkfi) [reply pkg/diff/diff.go:56]: Oh, I was just renaming it in this internal enum used by the remediator to be more consistent with the language used by the applier.

Comment (nan-yu) [reply pkg/diff/diff.go:56]: Gotcha.

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**